internal/controller/log.go

```diff
@@ -40,23 +43,28 @@ func (d *DeployLog) Init() error {
 
 // LogDeploy adds a record to the entries list.
 func (d *DeployLog) LogDeploy(timeStamp time.Time, podName string, podIP string, deploySuccessful bool, reason string) {
-	newEntry := logEntry{
+	newEntry := Entry{
 		TimeStamp:        timeStamp,
 		PodName:          podName,
 		PodIP:            podIP,
 		DeploySuccessful: deploySuccessful,
 		Reason:           reason,
 	}
 
-	d.Entries = append(d.Entries, newEntry)
+	for len(d.entries) >= d.maxEntries {
+		// Pull elements off the front of the slice to make sure that the newly appended record is one under the max record value.
+		d.entries = d.entries[1:]
+	}
+
+	d.entries = append(d.entries, newEntry)
 }
 
 // GetLog returns a json representation of the entries list.
-func (d *DeployLog) GetLog() []byte {
-	data, err := json.Marshal(d.Entries)
-	if err != nil {
-		log.Error("Could not marshal deploylog entries")
-	}
+func (d *DeployLog) GetLog() []Entry {
+	return d.entries
+}
 
-	return data
+// Len returns the number of records in the entries slice.
+func (d *DeployLog) Len() int {
```

Comment:
Is this `Len` function used somewhere ?

Reply:
Ah it was used for https://github.com/containous/maesh/pull/342/files#diff-6edd2e4ad97b4bf62fa0c917f59bc634R54 previously, but not anymore I suppose!

Reply:
Then I would remove it :)